node-manager/src/localstorage.rs

```diff
@@ -77,11 +71,20 @@ impl MutinyBrowserStorage {
         mnemonic
     }
 
-    pub(crate) fn get_mnemonic(&self) -> gloo_storage::Result<Mnemonic> {
-        let res: gloo_storage::Result<String> = self.get(mnemonic_key);
+    pub(crate) fn get_mnemonic(&self) -> anyhow::Result<Mnemonic> {
+        // TODO: here's another way to write this... but the error conversions end up being a pain in the ass
+        //
+        // self.get(mnemonic_key)
+        //     .and_then(|raw_mnemonic| {
+        //         Ok(Mnemonic::from_str(raw_mnemonic)
+        //             .with_context(|| format!("BIP 39 parse error"))?)
+        //     })
+        //     .with_context(|| format!("storage error"))
```

Comment:
I think if you use `.map` instead of `.and_then` then you don't need the extra `with_context`

Reply:
thanks I'll try that next time